### ISSUE TITLE
nv2 sign to support push

### ISSUE THREAD

**Issue by SteveLasker** [notaryproject/nv2#44]
```
nv2 sign -m x509 \
      -k ~/.ssh/wabbit-networks.key \
      -o net-monitor_v1-sbom.signature.jwt \
      --push registry.wabbit-networks.io/net-monitor \
      file:sbom_v1-manifest.json
```

**Comment by SteveLasker** [comment]
assigned to @shizhMSFT

**Comment by shizhMSFT** [comment]
New experience:
```bash
nv2 sign -m x509 -k example.key -c example.crt \
    --plain-http --push \
    oci://localhost:5000/test:latest
```

and

```bash
nv2 sign -m x509 -k example.key -c example.crt \
    --plain-http --push --push-reference oci://localhost:5000/test:latest \
    file:exported-manifest.json
```

**Comment by shizhMSFT** [comment]
Closed by e7592477e00f01b9bd5496dd6ba882c47ab77690
